### PR TITLE
 Azure RedisCache New Api v2023.08.01

### PR DESCRIPTION
**Pull request by koderjoker**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by wyunchi-ms** [comment]
Please re-record the test cases.

**Comment by wyunchi-ms** [comment]
image
Please fix the failed static analysis and test cases. You may need to re-record the failed test cases.

File: src/RedisCache/RedisCache/help/Clear-AzRedisCache.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.RedisCache.dll-Help.xml
+Module Name: Az.RedisCache
+online version:
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/help/Clear-AzRedisCache.md:4]:
```suggestion
online version:  https://learn.microsoft.com/powershell/module/az.rediscache/clear-azrediscache
```

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please fix the failed test cases

**Comment by koderjoker** [comment]
@msJinLei we're facing issues with updating the help text for new cmdlets with the following warning per each cmdlet. Could you please help us work around this?

`WARNING: [Update-MarkdownHelp] 02/20/2024 14:53:11 Command 'Get-AzRedisCacheAccessPolicy' not found in the session, skipping upgrade for  '..\RedisCache\help\Get-AzRedisCacheAccessPolicy.md'.`

File: src/RedisCache/RedisCache/help/Clear-AzRedisCache.md (hunk 1 above)
Comment (koderjoker) [reply src/RedisCache/RedisCache/help/Clear-AzRedisCache.md]: Added online version to all help files

File: src/RedisCache/RedisCache/Commands/ClearAzureRedisCache.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace Microsoft.Azure.Commands.RedisCache
+{
+    using Microsoft.Azure.Commands.RedisCache.Properties;
+    using ResourceManager.Common.ArgumentCompleters;
+    using System.Management.Automation;
+
+    [Cmdlet("Clear", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RedisCache", SupportsShouldProcess = true), OutputType(typeof(bool))]
+    public class ClearAzureRedisCache : RedisCacheCmdletBase
+    {
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/ClearAzureRedisCache.cs:23]: Please also provide resourceId and input object parameter sets as required https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#required-parameter-sets

File: src/RedisCache/RedisCache/Commands/GetAzureRedisCacheAccessPolicy.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Rest.Azure;
+    using ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.RedisCache.Models;
+
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RedisCacheAccessPolicy"), OutputType(typeof(PSRedisAccessPolicy))]
+    public class GetAzureRedisCacheAccessPolicy : RedisCacheCmdletBase
+    {
+        [Parameter(ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Name of resource group in which cache exists.")]
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/GetAzureRedisCacheAccessPolicy.cs:28]: Please also provide resourceId parameter set as required https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#required-parameter-sets

File: src/RedisCache/RedisCache/Commands/GetAzureRedisCacheAccessPolicyAssignment.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    using Microsoft.Azure.Commands.RedisCache.Models;
+    using Microsoft.Azure.Commands.RedisCache.Properties;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Rest.Azure;
+    using ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.RedisCache.Models;
+
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RedisCacheAccessPolicyAssignment"), OutputType(typeof(PSRedisAccessPolicyAssignment))]
+    public class GetAzureRedisCacheAccessPolicyAssignment : RedisCacheCmdletBase
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/GetAzureRedisCacheAccessPolicyAssignment.cs:26]: Please also provide resourceId parameter set as required https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#required-parameter-sets

File: src/RedisCache/RedisCache/Commands/RemoveAzureRedisCacheAccessPolicy.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+namespace Microsoft.Azure.Commands.RedisCache
+{
+    using ResourceManager.Common.ArgumentCompleters;
+    using System.Management.Automation;
+    using Properties;
+    using Models;
+
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RedisCacheAccessPolicy", DefaultParameterSetName = NormalParameterSet, SupportsShouldProcess = true), OutputType(typeof(bool))]
+    public class RemoveAzureRedisCacheAccessPolicy : RedisCacheCmdletBase
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/RemoveAzureRedisCacheAccessPolicy.cs:23]: Please also provide resourceId parameter set as required https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#required-parameter-sets

File: src/RedisCache/RedisCache/Commands/RemoveAzureRedisCacheAccessPolicyAssignment.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+namespace Microsoft.Azure.Commands.RedisCache
+{
+    using ResourceManager.Common.ArgumentCompleters;
+    using System.Management.Automation;
+    using Properties;
+    using Models;
+
+    [Cmdlet("Remove", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RedisCacheAccessPolicyAssignment", DefaultParameterSetName = NormalParameterSet, SupportsShouldProcess = true), OutputType(typeof(bool))]
+    public class RemoveAzureRedisCacheAccessPolicyAssignment : RedisCacheCmdletBase
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/RemoveAzureRedisCacheAccessPolicyAssignment.cs:23]: Please also provide resourceId parameter set as required https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#required-parameter-sets

File: src/RedisCache/RedisCache/Commands/NewAzureRedisCacheAccessPolicy.cs [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the access policy that is being added to the Redis cache.")]
+        [ValidateNotNullOrEmpty]
+        public string AccessPolicyName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "Permissions for the access policy. Learn how to configure permissions at https://aka.ms/redis/AADPreRequisites")]
+        [ValidateNotNullOrEmpty]
+        public string Permissions { get; set; }
```
Comment (msJinLei) [inline src/RedisCache/RedisCache/Commands/NewAzureRedisCacheAccessPolicy.cs:54]: Please use singular name https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/parameter-best-practices.md#singularity
```suggestion
        public string Permission { get; set; }
```

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/RedisCache/RedisCache/Commands/NewAzureRedisCacheAccessPolicy.cs (hunk 7 above)
Comment (koderjoker) [reply src/RedisCache/RedisCache/Commands/NewAzureRedisCacheAccessPolicy.cs]: Changed to Permission

File: src/RedisCache/RedisCache/Commands/RemoveAzureRedisCacheAccessPolicyAssignment.cs (hunk 6 above)
Comment (koderjoker) [reply src/RedisCache/RedisCache/Commands/RemoveAzureRedisCacheAccessPolicyAssignment.cs:23]: Required parameter sets provided

File: src/RedisCache/RedisCache/Commands/RemoveAzureRedisCacheAccessPolicy.cs (hunk 5 above)
Comment (koderjoker) [reply src/RedisCache/RedisCache/Commands/RemoveAzureRedisCacheAccessPolicy.cs:23]: Required parameter sets provided

File: src/RedisCache/RedisCache/Commands/GetAzureRedisCacheAccessPolicyAssignment.cs (hunk 4 above)
Comment (koderjoker) [reply src/RedisCache/RedisCache/Commands/GetAzureRedisCacheAccessPolicyAssignment.cs:26]: Required parameter sets provided

File: src/RedisCache/RedisCache/Commands/GetAzureRedisCacheAccessPolicy.cs (hunk 3 above)
Comment (koderjoker) [reply src/RedisCache/RedisCache/Commands/GetAzureRedisCacheAccessPolicy.cs]: Required parameter sets provided

File: src/RedisCache/RedisCache/Commands/ClearAzureRedisCache.cs (hunk 2 above)
Comment (koderjoker) [reply src/RedisCache/RedisCache/Commands/ClearAzureRedisCache.cs:24]: Required parameter sets provided

**Review by msJinLei — APPROVED** [review]

**Pull request merged**